recipes/date/all/conanfile.py

```diff
@@ -1,29 +1,84 @@
 import os
-from conans import ConanFile, tools
+from conans import ConanFile, CMake, tools
 
 
 class DateConan(ConanFile):
     name = "date"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/HowardHinnant/date"
     description = "A date and time library based on the C++11/14/17 <chrono> header"
-    topics = ("date", "datetime", "timezone", "calendar", "time", "iana-database")
+    topics = ("date", "datetime", "timezone",
+              "calendar", "time", "iana-database")
     license = "MIT"
-    exports_sources = ["patches/*"]
-    no_copy_source = True
+    exports_sources = ["patches/*", "CMakeLists.txt"]
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake", "cmake_find_package"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "use_system_tz_db": [True, False],
+               "use_tz_db_in_dot": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "use_system_tz_db": True,
+                       "use_tz_db_in_dot": False}
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
 
-    @property
-    def _source_subfolder(self):
-        return os.path.join(self.source_folder, "source_subfolder")
+    def configure(self):
+        #requires at least c++11 to work.
+        tools.check_min_cppstd(self, "11")
+
+        if self.settings.os == "Windows":
+            del self.options.use_system_tz_db
+            del self.options.use_tz_db_in_dot
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         extracted_dir = self.name + "-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["ENABLE_DATE_TESTING"] = False
+        if not self.settings.os == "Windows":
+            cmake.definitions["USE_TZ_DB_IN_DOT"] = self.options.use_tz_db_in_dot
+            cmake.definitions["USE_SYSTEM_TZ_DB"] = self.options.use_system_tz_db
+
+        cmake.configure()
+        return cmake
```

Comment:
To avoid doubling the times cmake is configured it's worth having a member to it.

```suggestion
        if self._cmake:
            return self._cmake
        self._cmake = CMake(self)
        cmake.definitions["ENABLE_DATE_TESTING"] = False
        if not self.settings.os == "Windows":
            self._cmake.definitions["USE_TZ_DB_IN_DOT"] = self.options.use_tz_db_in_dot
            self._cmake.definitions["USE_SYSTEM_TZ_DB"] = self.options.use_system_tz_db

        self._cmake.configure()
        return self._cmake
```

Reply:
Applied. thanks for hint - although its quite ugly ( requires initialization of _cmake = None )